File: other/tutorial/segmentation.py
import sys
import os
PathPop= ""
if os.path.isfile(PathPop+"population.py")==0:
    print "set the variable PathPop to the path where you compile population, for instance D:\Users/vtariel/Desktop/ANV/Population-build/. This folder must contain population.py"
    sys.exit(-1)
sys.path.append(PathPop)
from population import * 
	

img = Mat3UI8()
img.load(PathPop+"/image/rock3d.pgm")
#	img.loadFromDirectory("/home/vincent/Desktop/WorkSegmentation/sand/","500-755","pgm") #to load a stack of images
img = img(Vec3I32(0,0,0),Vec3I32(64,64,64))
img.display()
pde = PDE()
imgfilter= pde.nonLinearAnisotropicDiffusionDericheFast(img)
proc = Processing()
grain= proc.thresholdOtsuMethod(imgfilter)
grain.display()

visu = Visualization()
color= visu.labelForegroundBoundary(grain,img)
color.display()
grain.save(PathPop+"/image/grain.pgm")

scene = Scene3d()
visu.marchingCube(scene,grain)
visu.lineCube(scene,img)
scene.display()

File: other/tutorial/segmentation.cpp
#include"Population.h"//Single header
#include"data/notstable/CharacteristicCluster.h"
using namespace pop;//Population namespace

int main(int argc, char *argv[]){
    Mat3UI8 img;
    img.load(POP_PROJECT_SOURCE_DIR+std::string("/image/rock3d.pgm"));
    img = img(Vec3I32(0,0,0),Vec3I32(50,50,50));
    //        img.display();
    Mat3UI8 imgfilter=Processing::median(img,2,2);
    //        imgfilter.display();
    Mat2F32 m = Analysis::histogram(img);
    DistributionRegularStep d(m);
    //DistributionDisplay::display(d,d.getXmin(),d.getXmax());
    int threshold;
    Mat3UI8 grain= Processing::thresholdOtsuMethod(imgfilter,threshold);
    //        grain.display();
    Mat3RGBUI8 color= Visualization::labelForegroundBoundary(grain,imgfilter);
    color.display();
    grain.save("grain.pgm");
}

